File: src/test/resources/jython/flip.py
# CLIJ example ImageJ Ops jython: flip.py
#
# This script shows how stacks can be rotated in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

flipped = ops.run("CLIJ_flip", inputGPU, True, False, False)
# show result
result = ops.run("CLIJ_pull", flipped)
ui.show("flipped", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", flipped)
ops.run("CLIJ_close")

File: src/test/java/net/haesleinhuepf/clij/ops/examples/Flip.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_flip.CLIJ_flip;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class Flip {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif")
			.getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);
		// flip
		Object flipped = ij.op().run(CLIJ_flip.class, inputGPU, true, false, false);
		// show result
		Object result = ij.op().run(CLIJ_pull.class, flipped);
		ij.ui().show("flipped", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, flipped);


	}

	public static void main(String... args) throws IOException {
		Flip task = new Flip();
		task.run();
	}
}
